spec/controler_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'controler'
+require 'user_interface'
+
+RSpec.describe Controler do
+  let(:actions) { [double('NullAction')] }
+  let(:output) { StringIO.new }
+  let(:valid_input) { StringIO.new('1') }
+
+  it 'tells the user_interface to print the menu' do
+    ui = UserInterface.new(valid_input, output)
+    ctrl = Controler.new(ui, actions)
+
+    allow(actions[0]).to receive(:run)
+
+    ctrl.start
+
+    expect(output.string).to include(MENU_MESSAGE)
+  end
+
+  it 'receives the input from the user_interface' do
+    input = StringIO.new("2\n1\n")
+    ui = UserInterface.new(input, output)
+    ctrl = Controler.new(ui, actions)
+
+    allow(actions[0]).to receive(:run)
+
+    ctrl.start
+
+    expect(output.string).to include(ERROR_MESSAGE)
```

Comment:
you could make the user interface a double just for this test
so that the description of the test reflects what happens in the test more precisely.

then you could do:

```ruby
  it 'receives the input from the user interface' do
    ui = double(UserInterface, run: '1')
    controller = Controler.new(ui, actions)

    controller.start

    expect(ui).to have_received(:run).once
  end
```